config/utilities/uenv.py

```diff
@@ -61,23 +62,31 @@ def _get_uenvs():
         else:
             image_mount = _UENV_MOUNT_DEFAULT
 
-        inspect_cmd = f'{_UENV_CLI} image inspect {uenv_name} --format'
+        # Check if given uenv_name is a path to a squashfs archive
+        uenv_path = pathlib.Path(uenv_name)
+        if uenv_path.is_file():
+            # We cannot inspect for target systems
+            target_system = '*'
+            image_path = uenv_path
+            rfm_meta = image_path.parent / _RFM_META_DIR / _RFM_META
+        else:
+            inspect_cmd = f'{_UENV_CLI} image inspect {uenv_name} --format'
 
-        image_path = osext.run_command(
-            f"{inspect_cmd} '{{sqfs}}'", shell=True).stdout.strip()
-        target_system = osext.run_command(
-            f"{inspect_cmd} '{{system}}'", shell=True).stdout.strip()
+            image_path = osext.run_command(
+                f"{inspect_cmd} '{{sqfs}}'", shell=True).stdout.strip()
+            target_system = osext.run_command(
+                f"{inspect_cmd} '{{system}}'", shell=True).stdout.strip()
 
-        image_path = pathlib.Path(image_path)
+            image_path = pathlib.Path(image_path)
 
-        # FIXME temporary workaround for older uenv versions
-        if Version(uenv_version) >= Version('5.1.0-dev'):
-            meta_path = osext.run_command(
-                f"{inspect_cmd} '{{meta}}'", shell=True
-            ).stdout.strip()
-            rfm_meta = pathlib.Path(meta_path) / _RFM_META
-        else:
-            rfm_meta = image_path.parent / 'store.yaml'
+            # FIXME temporary workaround for older uenv versions
+            if Version(uenv_version) >= Version('5.1.0-dev'):
```

Comment:
I think that Eiger and Balfrin are the two systems that have uenv < 5.1 installed?

Can you update the comment to state that this can be removed when Eiger anc MCH upgrade?